src/test/java/pl/com/bottega/ecommerce/sales/domain/invoicing/BookKeeperTest.java

```diff
@@ -1,20 +1,146 @@
+        //Given
+        when(factory.create(any())).thenReturn(new Invoice(Id.generate(), null));
+        bookKeeper = new BookKeeper(factory);
+        invoiceRequest = new InvoiceRequestBuilder().addClient(clientData).build();
+
+        //When
+        Invoice invoice = bookKeeper.issuance(invoiceRequest, taxPolicy);
+
+        //Then
+        assertEquals(0, invoice.getItems().size());
+        assertEquals(Money.ZERO, invoice.getNet());
+        assertEquals(Money.ZERO, invoice.getGros());
+    }
+
+    @Test
+    void shouldReturnInvoiceWithNineItems() {
+        //Given
+        when(taxPolicy.calculateTax(any(), any())).thenReturn(tax);
+        when(factory.create(any())).thenReturn(new Invoice(Id.generate(), null));
+        bookKeeper = new BookKeeper(factory);
+        invoiceRequest = new InvoiceRequestBuilder().addClient(clientData).addItem(requestItem)
+                .addItem(requestItem).addItem(requestItem).addItem(requestItem).addItem(requestItem)
+                .addItem(requestItem).addItem(requestItem).addItem(requestItem).addItem(requestItem)
+                .build();
+
+        //When
+        Invoice invoice = bookKeeper.issuance(invoiceRequest, taxPolicy);
+
+        //Then
+        assertEquals(9, invoice.getItems().size());
+        assertEquals(Money.ZERO, invoice.getNet());
+        assertEquals(Money.ZERO, invoice.getGros());
+    }
+
+    @Test
+    void shouldCallCalculateMethodTwice() {
+        //Given
+        when(taxPolicy.calculateTax(any(), any())).thenReturn(tax);
+        when(factory.create(any())).thenReturn(new Invoice(Id.generate(), null));
+        bookKeeper = new BookKeeper(factory);
+        invoiceRequest = new InvoiceRequestBuilder().addClient(clientData).addItem(requestItem).addItem(requestItem).build();
 
+        //When
+        Invoice invoice = bookKeeper.issuance(invoiceRequest, taxPolicy);
+
+        //Then
+        verify(taxPolicy,times(2)).calculateTax(any(), any());
+    }
+
+    @Test
+    void shouldNotCallCalculateMethod() {
+        //Given
+        when(factory.create(any())).thenReturn(new Invoice(Id.generate(), null));
+        bookKeeper = new BookKeeper(factory);
+        invoiceRequest = new InvoiceRequestBuilder().addClient(clientData).build();
+
+        //When
+        Invoice invoice = bookKeeper.issuance(invoiceRequest, taxPolicy);
+
+        //Then
+        verify(taxPolicy,times(0)).calculateTax(any(), any());
     }
 
     @Test
-    void test() {
-        fail("not implemented");
+    void shouldCallCalculateMethodNineTimes() {
+        //Given
+        when(taxPolicy.calculateTax(any(), any())).thenReturn(tax);
+        when(factory.create(any())).thenReturn(new Invoice(Id.generate(), null));
+        bookKeeper = new BookKeeper(factory);
+        invoiceRequest = new InvoiceRequestBuilder().addClient(clientData).addItem(requestItem)
+                .addItem(requestItem).addItem(requestItem).addItem(requestItem).addItem(requestItem)
```

Comment:
jeżeli dodajemy coś wielokrotnie (pytanie czy to dobre podejście w testach, że zawsze mamy tę samą wartość) to można zrobić w builderze metodę o semantyce dodaj N razy